### PR TITLE
Feat/markdown richtext node

### PR DESCRIPTION
**Pull request by elizamaria**
Handles markdown-node element type
Renders markdown content as HTML content using the external package

### CODE REVIEW

File: packages/teleport-component-generator-react/src/react-mapping.ts [hunk 1]
```diff
@@ -21,6 +21,14 @@ export const ReactMapping: Mapping = {
         version: '2.2.1',
       },
     },
+    'markdown-node': {
+      elementType: 'Markdown',
```
Comment (JayaKrishnaNamburu) [inline packages/teleport-component-generator-react/src/react-mapping.ts:25]:
```js
elementType: 'markdown'
```

Please 😀

Comment (JayaKrishnaNamburu) [reply packages/teleport-component-generator-react/src/react-mapping.ts:25]: Should we build something like `dangerous-html` for this use-case too ?

**Comment by JayaKrishnaNamburu** [comment]
I think we can improve this a lot, instead of rendering markdown always to jsx at runtimes. We can use this
https://github.com/remarkjs/remark-rehype#remark-rehype
https://unifiedjs.com/explore/package/mdast-util-from-markdown/#frommarkdownvalue-encoding-options

And convert the markdown to `jsx` and `html` respectively. Users can write markdown, but there is no runtime on markdown to jsx. Because the way these values are changes only on the `studio` and nothing post build so this is most promising 😁 

Here is a example https://github.com/teleporthq/teleport-code-generators/pull/751/files#diff-d0233a558de7f5795a6759f3a72fb3b11f20aedf5fa077f54c32b3f34989a0aaR38

File: packages/teleport-component-generator-react/src/react-mapping.ts (hunk 1 above)
Comment (elizamaria) [reply packages/teleport-component-generator-react/src/react-mapping.ts:25]: This is how the element will look in jsx code, why with small letter? No need to build something new, this one works well.

Comment (JayaKrishnaNamburu) [reply packages/teleport-component-generator-react/src/react-mapping.ts:25]: Usually we go by, `elementType` with small case and then `semanticType` with the actual import or component name.

Yes, this works out of the box with `react` i am referring to all other frameworks and we can just do the. `mdx-to-jsx` at code-generation time. Without any runtime libraries 👍

**Comment by elizamaria** [comment]
> I think we can improve this a lot, instead of rendering markdown always to jsx at runtimes. We can use this https://github.com/remarkjs/remark-rehype#remark-rehype https://unifiedjs.com/explore/package/mdast-util-from-markdown/#frommarkdownvalue-encoding-options
> 
> And convert the markdown to `jsx` and `html` respectively. Users can write markdown, but there is no runtime on markdown to jsx. Because the way these values are changes only on the `studio` and nothing post build so this is most promising 😁
> 
> Here is a example https://github.com/teleporthq/teleport-code-generators/pull/751/files#diff-d0233a558de7f5795a6759f3a72fb3b11f20aedf5fa077f54c32b3f34989a0aaR38

Content will change after build, because this content can be updated from the CMS at any moment and the changes should be visible.

**Comment by JayaKrishnaNamburu** [comment]
ah, ok it's linked with CMS. Got it 👍

File: packages/teleport-component-generator-react/src/react-mapping.ts (hunk 1 above)
Comment (elizamaria) [reply packages/teleport-component-generator-react/src/react-mapping.ts:25]: for now we're focusing on publishing with next.js the things related to cms, if we will intend to support other frameworks we will indeed need to find another way. I will try with semnaticType

Comment (elizamaria) [reply packages/teleport-component-generator-react/src/react-mapping.ts:25]: JK I see we're using elementType for html-node, lottie-node as well, with uppercase, so this seems to be the common behavior. Also, if I'm having elementType together with semnaticType, elementType will be used in the generated jsx. I suggest leaving it as it is for now, what do you think?

Comment (JayaKrishnaNamburu) [reply packages/teleport-component-generator-react/src/react-mapping.ts:25]: yeah sure sure, no problem 👍

**Pull request merged**